File: src/algorithms/hash-maps/tracking/jewels-and-stones/sources/JewelsAndStones_test.java
public class JewelsAndStones_test {
    public static void main(String[] args) {
        assert JewelsAndStones.jewelsAndStones("aA", "aAAbbbb") == 3;
        assert JewelsAndStones.jewelsAndStones("z", "aAAbbbb") == 0;
        assert JewelsAndStones.jewelsAndStones("abc", "abcabc") == 6;
        assert JewelsAndStones.jewelsAndStones("aA", "") == 0;
        assert JewelsAndStones.jewelsAndStones("a", "a") == 1;
        assert JewelsAndStones.jewelsAndStones("a", "b") == 0;
        assert JewelsAndStones.jewelsAndStones("A", "aA") == 1;
        assert JewelsAndStones.jewelsAndStones("aa", "aaa") == 3;

        System.out.println("All tests passed!");
    }
}

File: src/algorithms/hash-maps/tracking/jewels-and-stones/__tests__/JewelsAndStones_test.cpp
#include "../sources/JewelsAndStones.cpp"
#include <cassert>
#include <iostream>

int main() {
    assert(jewelsAndStones("aA", "aAAbbbb") == 3);
    assert(jewelsAndStones("z", "aAAbbbb") == 0);
    assert(jewelsAndStones("abc", "abcabc") == 6);
    assert(jewelsAndStones("aA", "") == 0);
    assert(jewelsAndStones("a", "a") == 1);
    assert(jewelsAndStones("a", "b") == 0);
    assert(jewelsAndStones("A", "aA") == 1);
    assert(jewelsAndStones("aa", "aaa") == 3);

    std::cout << "All tests passed!" << std::endl;
    return 0;
}
